### PR TITLE
Correct rosdistro keys

### PR DESCRIPTION
**Pull request by ruffsl**
This corrects the rosdistro key names so rosdep can properly resolve and install missing dependencies, as well fix some issues in finding exported headers from upstream projects.

See official list of rosdistro keys here:
https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml

Related:
- https://github.com/borglab/gtsam/pull/173
- https://github.com/laurentkneip/opengv/pull/104
- https://github.com/dorian3d/DBoW2/pull/55
- https://github.com/opencv/opencv/pull/16322

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -9,16 +9,15 @@
...
   <buildtool_depend>cmake</buildtool_depend>
 
-  <depend>KimeraRPGO</depend>
-  <depend>GTSAM</depend>
-  <depend>gflags</depend>
-  <depend>glog</depend>
-  <depend>opencv</depend>
-  <depend>opengv</depend>
   <depend>dbow2</depend>
+  <depend>gtsam</depend>
```
Comment (ToniRV) [inline package.xml]: ok!

File: package.xml [hunk 2]
```diff
@@ -9,16 +9,15 @@
...
+  <depend>kimera_rpgo</depend>
```
Comment (ToniRV) [inline package.xml]: Not ok, as I think it is KimeraRPGO, we should make a PR to the RPGO repo updating the name there.

File: package.xml [hunk 3]
```diff
@@ -9,16 +9,15 @@
...
+  <depend>libgflags-dev</depend>
+  <depend>libgoogle-glog-dev</depend>
```
Comment (ToniRV) [inline package.xml]: I think the two above are irrelevant actually because we use cmake files to find glog/gflags as I had tons of issues with these two in multiple systems...

File: package.xml [hunk 4]
```diff
@@ -9,16 +9,15 @@
...
+  <depend>libopencv-dev</depend>
```
Comment (ToniRV) [inline package.xml]: Same, should be opencv  or rather `opencv3_catkin` if we use ETH's catkinized opencv3

**Comment by ToniRV** [comment]
So my understanding is that all the depend names in package.xml should reflect the ones that we want to depend on, which should be the ones we install in the workspace (+/- the ones that are already in ROS core packages).
So namings should follow:
https://github.com/MIT-SPARK/Kimera-VIO-ROS/blob/master/install/kimera_vio_ros_ssh.rosinstall

File: CMakeLists.txt [hunk 5]
```diff
@@ -145,6 +145,8 @@ target_include_directories(kimera_vio
   PUBLIC
     ${OpenCV_INCLUDE_DIRS}
     ${GTSAM_INCLUDE_DIR}
+    ${DBoW2_INCLUDE_DIRS}/..
+    ${opengv_DIR}/../../../include
```
Comment (ToniRV) [inline CMakeLists.txt]: Why is this needed? I would actually even argue that none of these should be needed as long as target_link_libraries is complete (this should set the include_directories automatically unless the dependencies are not exporting their includes correctly)

File: package.xml (hunk 4 above)
Comment (ToniRV) [reply package.xml]: If we use libopencv-dev, what are we assuming? That ppl installed opencv how?

File: package.xml (hunk 2 above)
Comment (ruffsl) [reply package.xml]: See https://github.com/MIT-SPARK/Kimera-RPGO/pull/45

File: package.xml (hunk 3 above)
Comment (ruffsl) [reply package.xml]: > use cmake files to find glog/gflags

Not sure being explicit about this external dependency the manifest would change what the CMake does to find the glog/gflags installed in the cmake path. Are you saying that the system install of glog/gflags in the latest ubuntu LTS is still an issue?

File: CMakeLists.txt (hunk 5 above)
Comment (ruffsl) [reply CMakeLists.txt]: > unless the dependencies are not exporting their includes correctly

There are a number of known issues back in catkin the would hide incorrect installation rules of headers given the melting pot that was the `devel` space. So some folks never needed to write correct install rules for exporting library artifacts. Now with colcon as the latest build tool for ROS workspaces, one needs to be more rigorous and explicit in writing install rules.

I'm not a CMake guru so perhaps you might spot a better fix here or in upstream packages:
* https://github.com/dorian3d/DBoW2/pull/55
* https://github.com/laurentkneip/opengv/pull/104

Just be sure to test it using a proper isolated cmake build workspace, like colcon does in the included Dockerfile.

File: package.xml (hunk 4 above)
Comment (ruffsl) [reply package.xml]: This is the correct/official rosdep key name for the opencv3 library:

https://github.com/ros/rosdistro/blob/4bfca115e18c733686b3b39524fbf1e227177574/rosdep/base.yaml#L2815

By declaring the same key name as the dependency, users with other packages depending on opencv3 don't have to monkey patch there manifest to use the same key to avoid rosdep from installing libopencv-dev from apt repo if the kimera workspace it builds from already includes a source build of opencv.

Note how this is used in the Dockerfile that builds the underlay for the ros wrapper:
https://github.com/MIT-SPARK/Kimera-VIO-ROS/pull/32/files#diff-3254677a7917c6c01f55212f86c57fbfR71-R95

File: CMakeLists.txt (hunk 5 above)
Comment (ruffsl) [reply CMakeLists.txt]: Lets move this to a new PR, I'd just like to get the rosdistro keys resolved.

**Comment by ToniRV** [comment]
@ruffsl if you want we can merge this one after updating to master and if Jenkins is ok with it :+1:

**Comment by ruffsl** [comment]
> @ruffsl if you want we can merge this one after updating to master and if Jenkins is ok with it +1

@ToniRV I've rebased this PR onto master. I think the 18.04 test might be flaky, as it passed before rebasing, and the error dosn't seem to pertain to ros-dep keys. The builds seems green though!
Could you also merge this PR as well: https://github.com/MIT-SPARK/Kimera-RPGO/pull/51
